exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/View.java

```diff
@@ -70,8 +72,9 @@ public boolean canModify() {
    *
    *  @throws IllegalStateException if the view is invalid (closed or nullptr)
    */
-  public long getViewNativeHandle() {
-    return super.getNativeHandle();
+  public NativeHandle getViewNativeHandle() {
+    checkState(nativeHandle.isValid(), "This handle is not valid: %s", this);
```

Comment:
This is code duplication and requires public `isValid` (which is also used by tests). Is there better solution?